src/main/java/com/auction/retailService/service/CustomerOrderService.java

```diff
@@ -0,0 +1,197 @@
+package com.auction.retailService.service;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.auction.retailService.constant.OrderStatusConstant;
+import com.auction.retailService.constant.RoleConstant;
+import com.auction.retailService.dto.CustomerOrderDTO;
+import com.auction.retailService.dto.CustomerOrderDetailDTO;
+import com.auction.retailService.dto.OrderPlacedDTO;
+import com.auction.retailService.dto.ProductDTO;
+import com.auction.retailService.dto.StoreProductsDTO;
+import com.auction.retailService.entity.CustomerOrder;
+import com.auction.retailService.entity.Order;
+import com.auction.retailService.entity.OrderProductsEntity;
+import com.auction.retailService.entity.StoreProduct;
+import com.auction.retailService.exception.OrderNotFoundException;
+import com.auction.retailService.exception.ProductOrderQuantityException;
+import com.auction.retailService.exception.StoreNotFoundException;
+import com.auction.retailService.exception.UserNotFoundException;
+import com.auction.retailService.exception.UserUnAuthorizedException;
+import com.auction.retailService.repository.CustomerOrderRepository;
+import com.auction.retailService.repository.OrderProductsRepository;
+
+@Service
+public class CustomerOrderService {
+	
+	@Autowired
+	CustomerOrderRepository costOrderRepositoty;
+	
+	@Autowired
+	OrderProductsRepository orderProductRepo;
+	
+	@Autowired
+	ProductService productService;
+	
+	@Autowired
+	StoreProductService storeProductservice;
+	
+	@Autowired
+	UserService userService;
+	
+	@Autowired
+	StoreService storeservice;
+	
+	
+	@Transactional
+	public OrderPlacedDTO addOrder(CustomerOrderDTO orderDto, Long orderBy) {
+		OrderPlacedDTO dto = new OrderPlacedDTO();
+		CustomerOrder order = new CustomerOrder();
+		order.setStoreId(orderDto.getStoreId());
```

Comment:
This is conversion logic. Keep your service with a  single responsibility.

Reply:
furthermore, if you need to set several attributes to a dto, you could use a builder